__tests__/app.test.js

```diff
@@ -114,36 +114,92 @@ describe("GET /api/articles", () => {
             created_at: expect.any(String),
             votes: expect.any(Number),
             article_img_url: expect.any(String),
-            comment_count: expect.any(String)
+            comment_count: expect.any(Number),
           });
-          expect(article).not.toHaveProperty('body')
+          expect(article).not.toHaveProperty("body");
         });
       });
   });
   it("comment_count provides the right ammount of comments", () => {
     return request(app)
       .get("/api/articles")
       .then((res) => {
-        const articles = res.body.articles
-        articles.forEach(article => {
+        const articles = res.body.articles;
+        articles.forEach((article) => {
           let count = 0;
-          testData.commentData.forEach(comment => {
+          testData.commentData.forEach((comment) => {
             if (comment.article_id === article.article_id) {
-              count++
+              count++;
             }
-          })
-          expect(article.comment_count).toBe(count.toString())
-        })
-      })
-  })
-  it('articles array is sorted by date in descending order', () => {
+          });
+          expect(article.comment_count).toBe(count);
+        });
+      });
+  });
+  it("articles array is sorted by date in descending order", () => {
     return request(app)
       .get("/api/articles")
       .then((res) => {
-        const articles = res.body.articles
-        expect(articles).toBeSortedBy("created_at", {descending: true})
-      })
-  })
+        const articles = res.body.articles;
+        expect(articles).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+});
+
+describe("GET /api/articles/:article_id/comments", () => {
+  it("responds with an array of comments for the given article_id. All comments have respective correct properties. Status code: 200", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((res) => {
+        const comments = res.body.comments;
+        comments.forEach((comment) => {
```

Comment:
We should be checking the length of the comments that are returned before we run assertions on each one. 

We need to be really careful here as if an empty array were to come back from the database, the callback function of the forEach would not be invoked and the assertion inside would never be read. 

This would lead to a false positive test